src/test/java/org/opensearch/security/auditlog/sink/WebhookAuditLogTest.java

```diff
@@ -774,6 +774,162 @@ private SSLContext createSSLContext() throws Exception {
         return sslContext;
     }
 
+    @Test
+    public void basicAuthPostTest() throws Exception {
+        TestHttpHandler handler = new TestHttpHandler();
+
+        int port = findFreePort();
+        server = ServerBootstrap.bootstrap()
+            .setListenerPort(port)
+            .setHttpProcessor(HttpProcessors.server("Test/1.1"))
+            .setRequestRouter((request, context) -> handler)
+            .create();
+
+        server.start();
+
+        String url = "http://localhost:" + port + "/endpoint";
+        String username = "test_user";
+        String password = "test_password";
+
+        // Test with basic auth credentials - POST JSON
+        Settings settings = Settings.builder()
+            .put("plugins.security.audit.config.webhook.url", url)
+            .put("plugins.security.audit.config.webhook.format", "json")
+            .put("plugins.security.audit.config.username", username)
+            .put("plugins.security.audit.config.password", password)
+            .put("path.home", ".")
+            .put(
+                SSLConfigConstants.SECURITY_SSL_TRANSPORT_TRUSTSTORE_FILEPATH,
+                FileHelper.getAbsoluteFilePathFromClassPath("auditlog/truststore.jks")
+            )
+            .build();
+
+        LoggingSink fallback = new LoggingSink("test", Settings.EMPTY, null, null);
+        WebhookSink auditlog = new WebhookSink("name", settings, ConfigConstants.SECURITY_AUDIT_CONFIG_DEFAULT, null, fallback);
+        AuditMessage msg = MockAuditMessageFactory.validAuditMessage();
+        auditlog.store(msg);
+
+        // Verify request was made
+        assertThat(handler.method, is("POST"));
+        Assert.assertNotNull(handler.body);
+        Assert.assertTrue(handler.body.contains("{"));
```

Comment:
nit: This seems like a weak assertion compared to the next one which checks for keys and values.